File: PRACTICA PRIMER PARCIAL/Ejercicio28/Python/KEsimoDigito.py
def obtener_kesimo_digito(n, k):
    num_str = str(n)
    cantidad_digitos = len(num_str)
    k_ésimo_dígito = int(num_str[k - 1])
    return cantidad_digitos, k_ésimo_dígito

# Lectura de entrada
n, k = map(int, input("Ingrese el número N y el dígito K separados por un espacio: ").split())


cantidad_digitos, k_ésimo_dígito = obtener_kesimo_digito(n, k)

# Salida
print("Cantidad de dígitos:", cantidad_digitos)
print("El dígito", k, "es:", k_ésimo_dígito)

File: PRACTICA PRIMER PARCIAL/Ejercicio28/Java/KEsimoDigito.java
import java.util.Scanner;

public class KEsimoDigito {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Lectura de entrada
        System.out.print("Ingrese el número N y el dígito K separados por un espacio: ");
        long n = scanner.nextLong();
        int k = scanner.nextInt();

        
        long[] resultados = obtenerKesimoDigito(n, k);
        long cantidadDigitos = resultados[0];
        long kEsimoDigito = resultados[1];

        
        System.out.println("Cantidad de dígitos: " + cantidadDigitos);
        System.out.println("El dígito " + k + " es: " + kEsimoDigito);
    }

    public static long[] obtenerKesimoDigito(long n, int k) {
        String numStr = Long.toString(n);
        long cantidadDigitos = numStr.length();
        long kEsimoDigito = Character.getNumericValue(numStr.charAt(k - 1));
        return new long[]{cantidadDigitos, kEsimoDigito};
    }
}
